libcxx/include/__tree

```diff
@@ -2030,10 +2035,12 @@ template <class _Key>
 typename __tree<_Tp, _Compare, _Allocator>::size_type
 __tree<_Tp, _Compare, _Allocator>::__count_unique(const _Key& __k) const {
   __node_pointer __rt = __root();
+  auto __comp = __three_way_comparator<value_compare, _Key, value_type>(value_comp());
```

Comment:
I'd like to find a way to express the fact that this isn't just a normal three-way comparison. I think that changing the name to e.g. `__hoisted_three_way_comparator` or `__lazy_three_way_comparator` (which isn't quite right though) would help.

Reply:
It turns out that we have something quite similar called `__synth_three_way`. I went for `__lazy_synth_three_way_comparator` for now. It's still not quite right, but probably close enough. `__maybe_lazy_potentially_synthesized_three_way_comparator` seems a bit long.